pkg/daemon/update.go

```diff
@@ -312,6 +333,38 @@ type MachineConfigDiff struct {
 	units    bool
 }
 
+// See if this was Rust we wouldn't have nullable pointers by default...
+func canonicalizeStringArray(a []string) []string {
```

Comment:
btw this isn't an array it's a slice.  isn't nil slice []somename vs empty canonical?

Reply:
you dont really need this anyway see above.

Reply:
Removed this function in latest push

Reply:
yay!